### ISSUE TITLE
[Bug]: parserOptions.ecmaFeatures.jsx not set when using presets

### ISSUE THREAD

**Issue by CleyFaye** [jsx-eslint/eslint-plugin-react#3523]
### Is there an existing issue for this?

- [X] I have searched the existing issues and my issue is unique
- [X] My issue appears in the command-line and not only in the text editor

### Description Overview

When extending eslint configuration with `"extends": ["plugin:react/recommended"]`, the `parserOptions` are not set to enable JSX support.

According to the README, `parserOptions.ecmaFeatures.jsx` should be set to `true` by all presets.

```
% cat .eslintrc.cjs
module.exports = {
  parserOptions: {
    ecmaVersion: "latest",
    sourceType: "module"
  },
  extends: [
    "eslint:recommended",
    "plugin:react/all"
  ],
  settings: {
    react: {
      version: "detect"
    }
  }
}

% npx eslint --print-config somefile.js | jq .parserOptions
{
  "ecmaVersion": "latest",
  "sourceType": "module"
}
```

This property not being set breaks parsing of JavaScript files containing JSX.

### Expected Behavior

According to the documentation:

> Note: These configurations will import eslint-plugin-react and enable JSX in [parser options](https://eslint.org/docs/user-guide/configuring/language-options#specifying-parser-options).

But using the presets (all and recommended) do not set this property.

It used to work in v7.31.11 but is broken in v7.32.1, which is actually tagged "latest" on npmjs.

With v7.31.11, the example in the overview above yields this result:

```
% npx eslint --print-config somefile.js | jq .parserOptions
{
  "ecmaVersion": "latest",
  "sourceType": "module",
  "ecmaFeatures": {
    "jsx": true
  }
}
```

and eslint works fine with files containing JSX.

### eslint-plugin-react version

v7.32.1

### eslint version

v8.32.0

### node version

v18.13.0

**Comment by ljharb** [comment]
I would suggest using the airbnb config, instead of the "recommended" config (which isn't actually recommended, because it's very out of date due to semver constraints).

**Comment by ljharb** [comment]
That said, it's currently set here: https://github.com/jsx-eslint/eslint-plugin-react/blob/master/configs/all.js#L29-L35 for the flat config, so we should also be setting it for the legacy config, assuming the flat config won't error on it.

**Comment by lkraav** [comment]
7.34.0: somehow this flat config only applies `parserOptions.ecmaFeatures.jsx` as a standalone config object, but not when spreading inside another. Is this a bug or a feature?

```js
// @see https://eslint.org/docs/latest/use/configure/migration-guide#predefined-and-shareable-configs
import js from "@eslint/js";
import reactRecommended from "eslint-plugin-react/configs/recommended.js";

export default [
  js.configs.recommended,
  // reactRecommended, // `parserOptions.ecmaFeatures.jsx` is set, but I don't want it as a global config objet?
  {
    files: ["apps/admin/**/*.{js,jsx}"],
    ...reactRecommended, // `parserOptions.ecmaFeatures.jsx` is missing.
  },
];

**Comment by ljharb** [comment]
That's likely because `languageOptions` is non-enumerable, so it won't spread.

**Comment by lkraav** [comment]
> That's likely because `languageOptions` is non-enumerable, so it won't spread.

Yeah I suspected this being related after inspecting the code.

For a new person, this distinction might be a bit confusing, because https://github.com/jsx-eslint/eslint-plugin-react/blob/2e6b5578cf9d8df2a4d35b9f5f3d851cf07ff483/README.md?plain=1#L231-L255 doesn't really say it will be mandatory to also spread `languageOptions` manually to get `jsx` parsed.

**Comment by ljharb** [comment]
That's true. I think long term we need separate entry points for flat config, and until we have those, this plugin should only be used with flat config in concert with eslint's flat compat wrapper.
